src/libbson/tests/test-string.c

```diff
@@ -356,9 +357,73 @@ test_bson_string_capacity (void *unused)
       large_str[UINT32_MAX - 2u] = 's'; // Restore.
    }
 
+   // Test allocating the largest possible string.
+   {
+      bson_string_t *str = _bson_string_alloc (2147483648);
+      ASSERT_CMPUINT32 (str->alloc, ==, UINT32_MAX);
+      ASSERT_CMPUINT32 (str->len, ==, 0);
+      bson_string_free (str, true);
+   }
+
    bson_free (large_str);
 }
 
+static void
+test_bson_string_append_ex (void)
+{
+   bson_string_t *str;
+   char *s;
+
+   str = bson_string_new (NULL);
+   bson_string_append_ex (str, "the quick brown fox jumps over the lazy dog", 10);
+   s = bson_string_free (str, false);
+   BSON_ASSERT (s);
+   BSON_ASSERT (!strcmp (s, "the quick "));
```

Comment:
```suggestion
   ASSERT_CMPSTR (s, "the quick ");
```
Use `ASSERT` and `ASSERT_CMPSTR` in test code.